storage/innobase/buf/buf0flu.cc

```diff
@@ -2135,10 +2168,19 @@ static os_thread_ret_t DECLARE_THREAD(buf_flush_page_cleaner)(void*)
     const double dirty_pct= double(dirty_blocks) * 100.0 /
       double(UT_LIST_GET_LEN(buf_pool.LRU) + UT_LIST_GET_LEN(buf_pool.free));
 
+    /* idle_flush steps kicks in
+    - if page cleaner is idle
+    - there are dirty pages (dirty_pct < innodb_max_dirty_pages_pct_lwm)
+    - server is idle (no trx_t::commit() activity)
+    - there are no pending reads (seen with io-bound workload) */
+    const bool idle_flush= buf_pool.page_cleaner_idle() &&
+      dirty_blocks != 0 && !srv_check_activity(&last_activity_count) &&
+      (buf_pool.n_pend_reads == 0);
```

Comment:
The `dirty_blocks != 0 &&` is redundant here, because we just checked for `!dirty_blocks` a couple of statements earlier.

I think that we are computing this unnecessarily if `srv_max_dirty_pages_pct_lwm` is at its default value 0. The following should avoid that:
```c++
    bool idle_flush= false;

    if (lsn_limit);
    else if (srv_max_dirty_pages_pct_lwm != 0.0)
    {
      const ulint activity_count= srv_get_activity_count();
      if (activity_count != last_activity_count)
        last_activity_count= activity_count;
      else if (buf_pool.page_cleaner_idle() && buf_pool.n_pending_reads == 0)
      {
        idle_flush= true;
        buf_pool.update_last_activity_count(activity_count);
      }
      if (!idle_flush && dirty_pct < srv_max_dirty_pages_pct_lwm)
        goto unemployed;
    }
```
Note: This includes `buf_pool.update_last_activity_count()`, and I would pass the count as a parameter, to avoid a redundant call.

Reply:
done